tests/src/test/scala/com/nvidia/spark/rapids/UrlFunctionsSuite.scala

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+class UrlFunctionsSuite extends SparkQueryCompareTestSuite {
+  def validUrlEdgeCasesDf(session: SparkSession): DataFrame = {
+    import session.sqlContext.implicits._
+    // [In search of the perfect URL validation regex](https://mathiasbynens.be/demo/url-regex)
+    Seq[String](
+      "http://foo.com/blah_blah",
+      "http://foo.com/blah_blah/",
+      "http://foo.com/blah_blah_(wikipedia)",
+      "http://foo.com/blah_blah_(wikipedia)_(again)",
+      "http://www.example.com/wpstyle/?p=364",
+      "https://www.example.com/foo/?bar=baz&inga=42&quux",
+      "http://✪df.ws/123",
+      // "http://userid:password@example.com:8080",
```

Comment:
Remove these if we don't want.

Reply:
They are failing tests now, will uncomment them after the kernel fixes these cases.